File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

int z_count = 0;
int count = 0;

string to_binary(int num) {
	string s = "";
	while (num > 0) {
		if (num % 2 == 1) s = "1" + s;
		else s = "0" + s;
		num >>= 1;
	}
	return s;
}
void convert(string s)
{
    if (s == "1")
    {
        return;
    }
    int tmp = 0;
    string change_string = "";
    for (auto i : s)
    {
       if (i == '0')
       {
           z_count++;
           tmp++;
       }
    }
    int erase_0 = s.size() - tmp;
    change_string = to_binary(erase_0);
    count++;
    convert(change_string);
}
vector<int> solution(string s) {
    convert(s);
    vector<int> answer;
    answer.push_back(count);
    answer.push_back(z_count);
    return answer;
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def binary(s):
    binaryNum = format(s, 'b')
    return binaryNum[2:]
def solution(s):
    answer = []
    count_zero = 0
    count = 0
    while(1):
        count+=1
        count_zero += s.count("0")
        s = format(len(s) - s.count("0"),'b')
        if(s == "1"):
            break
    answer.append(count)
    answer.append(count_zero)
    return answer